WordPress/openverse-api
# Add sentry

## Fixes


Fixes #399  by @sarayourfriend 

## Description


Adds basic Sentry monitoring to the API to capture unhandled exceptions. This is the bare minimum monitoring we should have.

Something is causing the dead-link filters integration test to fail. I'm not sure how to fix this, hopefully someone else can help debug the issue. It might have to do with DRF being upgraded from 3.12.x to 3.13.0 due to installing the `sentry-sdk` library.

## Testing Instructions

The only way to properly test this is to checkout this branch and run `just recreate && just init` then cause an error to happen (easiest way is through a bad code change that will only cause a runtime error) and verify it shows in the Sentry dashboard. The dashboard is private so if you need access to it ping me and I'll share how to access it.

You'll also need to run the app ~in~ without DEBUG mode or change the conditional in settings.py to allow Sentry to run locally. Of course we don't want Sentry errors to populate outside of production normally.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
